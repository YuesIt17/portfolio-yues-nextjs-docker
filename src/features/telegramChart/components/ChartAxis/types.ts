import {TChartDataLabel} from '@/utils/types';

export type TChartAxis = {
  points: string;
  typeAxis: 'x' | 'y';
  labels: TChartDataLabel[];
  labelPaddingX?: number;
  labelPaddingY?: number;
  maxDataX?: number;
};
