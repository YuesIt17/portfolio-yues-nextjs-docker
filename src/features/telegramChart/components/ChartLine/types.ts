export type TChartLine = {
  points: string;
  fill?: string;
  stroke?: string;
  isVisible?: boolean;
};
