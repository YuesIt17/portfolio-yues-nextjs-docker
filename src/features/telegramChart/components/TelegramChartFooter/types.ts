import {TChartDataLine} from '@/utils/types';

export type TTelegramChartFooter = {
  data: TChartDataLine[] | undefined;
  onChange: (line: TChartDataLine, isVisible: boolean) => void;
};
