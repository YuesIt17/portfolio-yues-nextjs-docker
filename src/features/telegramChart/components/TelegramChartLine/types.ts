import {TChartDataLine, TChartPrepareData} from '@/utils/types';

export type TTelegramChartLine = {
  data: TChartDataLine[];
  width?: number;
  peddingSize?: number;
} & Omit<TChartPrepareData, 'lines'>;
