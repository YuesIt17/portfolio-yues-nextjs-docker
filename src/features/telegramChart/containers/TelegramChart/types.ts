export type TTelegramChart = {
  width: number;
};

export type TTelegramChartStyle = TTelegramChart;
