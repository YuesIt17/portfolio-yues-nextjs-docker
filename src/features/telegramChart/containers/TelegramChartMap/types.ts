import {TChartDataLine} from '@/utils/types';

export type TTelegramChartMap = {
  width: number;
  data: TChartDataLine[];
  maxDataX: number;
  maxDataY: number;
};

export type TTelegramChartMapStyle = {
  width: number;
};
