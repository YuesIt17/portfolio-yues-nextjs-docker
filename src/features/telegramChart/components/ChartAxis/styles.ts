import {css, Theme} from '@emotion/react';
import {FONT_SIZE} from '../../constants';

export const useStyles = () => ({
  line: (theme: Theme) =>
    css({
      stroke: theme.colors.secondary.dark,
      strokeWidth: 0.5,
    }),
  label: (theme: Theme) =>
    css({
      fontSize: FONT_SIZE,
      fill: theme.colors.secondary.dark,
    }),
});
