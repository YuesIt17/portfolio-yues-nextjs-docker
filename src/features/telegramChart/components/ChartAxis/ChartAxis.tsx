import React, {memo, useCallback} from 'react';
import {TChartDataLabel} from '@/utils/types';
import {FONT_SIZE} from '../../constants';
import {useStyles} from './styles';
import {TChartAxis} from './types';

export const ChartAxis = memo(
  ({
    points,
    typeAxis,
    labelPaddingX = 0,
    labelPaddingY = 0,
    labels,
    maxDataX,
  }: TChartAxis) => {
    const styles = useStyles();
    const coordinateY = useCallback(
      (item?: TChartDataLabel) =>
        typeAxis === 'x'
          ? labelPaddingY + FONT_SIZE * 2
          : item && item.coordinate >= FONT_SIZE
          ? item.coordinate - FONT_SIZE
          : 0,
      [typeAxis, labelPaddingY]
    );

    return (
      <g>
        <polyline css={styles.line} points={points} />
        {typeAxis === 'x' &&
          labels.map((item, index) => (
            <text
              key={index}
              x={item.coordinate}
              y={coordinateY()}
              css={styles.label}
            >
              {item.label}
            </text>
          ))}
        {typeAxis === 'y' &&
          labels.map((item, index) => (
            <g key={index}>
              <text x={labelPaddingX} y={coordinateY(item)} css={styles.label}>
                {item.label}
              </text>
              <polyline
                css={styles.line}
                points={`${labelPaddingX},${item.coordinate} ${maxDataX},${item.coordinate}`}
              />
            </g>
          ))}
      </g>
    );
  }
);

ChartAxis.displayName = 'ChartAxis';
