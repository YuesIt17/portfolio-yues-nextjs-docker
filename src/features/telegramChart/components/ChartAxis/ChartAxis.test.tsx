import React from 'react';
import {themeTestRenderer} from '@/utils/tests';
import {ChartAxis} from './ChartAxis';
import {TChartAxis} from './types';

const axisParams = {
  labels: [
    {label: '1', coordinate: 10},
    {label: '2', coordinate: 20},
  ],
  points: '1,1 2,2',
  labelPaddingX: 10,
  labelPaddingY: 10,
  maxDataX: 10,
} as TChartAxis;

describe('ChartAxis test', () => {
  test('Check render ChartAxis X component', () => {
    const chartAxisX = themeTestRenderer(
      <ChartAxis {...axisParams} typeAxis="x" />
    ).toJSON();

    expect(chartAxisX).toMatchSnapshot();
  });

  test('Check render ChartAxis Y component', () => {
    const chartAxisY = themeTestRenderer(
      <ChartAxis {...axisParams} typeAxis="y" />
    ).toJSON();

    expect(chartAxisY).toMatchSnapshot();
  });
});
