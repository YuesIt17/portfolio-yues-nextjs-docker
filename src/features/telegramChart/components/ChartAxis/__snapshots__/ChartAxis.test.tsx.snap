// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartAxis test Check render ChartAxis X component 1`] = `
<g>
  <polyline
    className="css-1ied4zn-line"
    points="1,1 2,2"
  />
  <text
    className="css-1nvft8-label"
    x={10}
    y={34}
  >
    1
  </text>
  <text
    className="css-1nvft8-label"
    x={20}
    y={34}
  >
    2
  </text>
</g>
`;

exports[`ChartAxis test Check render ChartAxis Y component 1`] = `
<g>
  <polyline
    className="css-1ied4zn-line"
    points="1,1 2,2"
  />
  <g>
    <text
      className="css-1nvft8-label"
      x={10}
      y={0}
    >
      1
    </text>
    <polyline
      className="css-1ied4zn-line"
      points="10,10 10,10"
    />
  </g>
  <g>
    <text
      className="css-1nvft8-label"
      x={10}
      y={8}
    >
      2
    </text>
    <polyline
      className="css-1ied4zn-line"
      points="10,20 10,20"
    />
  </g>
</g>
`;
